src/main/kotlin/org/rust/ide/inspections/RsWithMacrosInspectionVisitor.kt

```diff
@@ -0,0 +1,175 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections
+
+import org.rust.lang.core.macros.prepareForExpansionHighlighting
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsAttrProcMacroOwner
+import org.rust.stdext.removeLast
+
+/**
+ * This is a non-recursive element visitor, but if it faces a macro invocation, it accepts all elements
+ * expanded from the macro. This visitor is intended to be used in [RsLocalInspectionTool] implementations.
+ */
+abstract class RsWithMacrosInspectionVisitor : RsVisitor() {
```

Comment:
Consider the following code:

```rust
#[attr_as_is]
fn func() {
    bar(1);
}
fn bar() {}
```

Looks like [`checkValueArgumentList`](https://github.com/intellij-rust/intellij-rust/blob/2ee7de7cc65b5b8362b3fedcdf63b74ec2996907/src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt#L1153) will be called twice for different PSI elements (one expanded, one not). Is it expected?